velox/functions/prestosql/tests/ArrayFrequencyTest.cpp

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/functions/prestosql/tests/utils/FunctionBaseTest.h"
+
+using namespace facebook::velox;
+using namespace facebook::velox::test;
+
+namespace {
+class ArrayFrequencyTest : public functions::test::FunctionBaseTest {
+ protected:
+  // Evaluate an expression.
+  void testExpr(
+      const VectorPtr& expected,
+      const std::string& expression,
+      const std::vector<VectorPtr>& input) {
+    auto result = evaluate<BaseVector>(expression, makeRowVector(input));
+    assertEqualVectors(expected, result);
+  }
+};
+} // namespace
+
+TEST_F(ArrayFrequencyTest, integerArray) {
+  auto array = makeNullableArrayVector<int64_t>(
+      {{2, 1, 1, -2},
+       {},
+       {1, 2, 1, 1, 1, 1},
+       {-1, std::nullopt, -1, -1},
+       {std::numeric_limits<int64_t>::max(),
+        std::numeric_limits<int64_t>::max(),
+        1,
+        std::nullopt,
+        0,
+        1,
+        std::nullopt,
+        0}});
+
+  auto expectedKeys = makeFlatVector<int64_t>(
+      {1,
+       2,
+       -2,
+       // empty key
+       1,
+       2,
+       -1,
+       std::numeric_limits<int64_t>::max(),
+       1,
+       0});
+  auto expectedValues = makeFlatVector<int>(
+      {2,
+       1,
+       1,
+       // empty value
+       5,
+       1,
+       3,
+       2,
+       2,
+       2});
+  auto expected = makeMapVector({0, 3, 3, 5, 6}, expectedKeys, expectedValues);
```

Comment:
Any particular reason not to use

```
  // Create map vector from nested std::vector representation.
  template <typename TKey, typename TValue>
  MapVectorPtr makeMapVector(
      const std::vector<std::vector<std::pair<TKey, std::optional<TValue>>>>&
          maps,
      const TypePtr& type =
          MAP(CppToType<TKey>::create(), CppToType<TValue>::create())) {
```

Reply:
somehow missed that api, definitely much cleaner